cmd/cmd.go

```diff
@@ -67,31 +65,34 @@ func ExecInCoreSnap() {
 
 	full := filepath.Join(newCore, exe)
 	if !osutil.FileExists(full) {
-		if rev, err := os.Readlink(oldCore); err != nil {
-			return
-		} else if revno, err := strconv.Atoi(rev); err != nil || revno < minOldRevno {
-			return
-		}
-
 		full = filepath.Join(oldCore, exe)
 		if !osutil.FileExists(full) {
 			return
 		}
 	}
 
-	// ensure we do not re-exec into an older version of snapd
-	currentSnapd, err := os.Stat(exe)
-	if err != nil {
-		logger.Noticef("cannot stat %q: %s", exe, err)
-		return
+	// ensure we do not re-exec into an older version of snapd, look
+	// for info file and ignore version of core that do not yet have
+	// it
+	fullInfo := filepath.Join(newCore, "/usr/lib/snapd/info")
```

Comment:
can't we remember which one we used for finding full inside a core, instead of trying again this way here?